test/e2e/storage/csi_mock_volume.go

```diff
@@ -2074,27 +2109,95 @@ var _ = utils.SIGDescribe("CSI mock volume", func() {
+				ginkgo.By("Checking the first pod mount options")
+				framework.ExpectEqual(nodeStageMountOpts, t.expectedFirstMountOptions, "Expect NodeStageVolumeRequest.VolumeCapability.MountVolume. to equal %q; got: %q", t.expectedFirstMountOptions, nodeStageMountOpts)
+				framework.ExpectEqual(nodePublishMountOpts, t.expectedFirstMountOptions, "Expect NodePublishVolumeRequest.VolumeCapability.MountVolume.VolumeMountGroup to equal %q; got: %q", t.expectedFirstMountOptions, nodeStageMountOpts)
+
+				ginkgo.By("Checking the CSI driver calls for the first pod")
+				gomega.Expect(unstageCalls.Load()).To(gomega.BeNumerically("==", 0), "NodeUnstage was unexpectedly called for the first pod")
+				gomega.Expect(unpublishCalls.Load()).To(gomega.BeNumerically("==", 0), "NodeUnpublish was unexpectedly called for the first pod")
+				gomega.Expect(stageCalls.Load()).To(gomega.BeNumerically(">", 0), "NodeStage was not called for the first pod")
+				gomega.Expect(publishCalls.Load()).To(gomega.BeNumerically(">", 0), "NodePublish was not called for the first pod")
+
+				if !t.startSecondPod {
+					return
+				}
+
+				ginkgo.By("Starting the second pod")
+				// Skip scheduler, it would block scheduling the second pod with ReadWriteOncePod PV.
+				pod, err = m.cs.CoreV1().Pods(pod.Namespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err, fmt.Sprintf("get first pod"))
+				nodeSelection := e2epod.NodeSelection{Name: pod.Spec.NodeName}
+				pod2, err := startPausePodWithSELinuxOptions(f.ClientSet, claim, nodeSelection, f.Namespace.Name, t.secondPodSELinuxOpts)
+				framework.ExpectNoError(err, "Failed to create second pause pod with SELinux context %s: %v", t.secondPodSELinuxOpts, err)
+				m.pods = append(m.pods, pod2)
+
+				ginkgo.By("Waiting for the second pod to fail to start because of RWOP")
+				eventSelector := fields.Set{
+					"involvedObject.kind":      "Pod",
+					"involvedObject.name":      pod2.Name,
+					"involvedObject.namespace": pod2.Namespace,
+					"reason":                   events.FailedMountVolume,
+				}.AsSelector().String()
+				var msg string
+				if t.expectedUnstage {
+					// This message is emitted before kubelet checks for ReadWriteOncePod
+					msg = "conflicting SELinux labels of volume"
+				} else {
+					msg = "volume uses the ReadWriteOncePod access mode and is already in use by another pod"
+				}
+				err = e2eevents.WaitTimeoutForEvent(m.cs, pod2.Namespace, eventSelector, msg, f.Timeouts.PodStart)
+				framework.ExpectNoError(err, "while waiting for event in the second test pod:", msg)
+
+				// count fresh CSI driver calls between the first and the second pod
+				nodeStageMountOpts = nil
+				nodePublishMountOpts = nil
+				unstageCalls.Store(0)
+				unpublishCalls.Store(0)
+				stageCalls.Store(0)
+				publishCalls.Store(0)
+
+				ginkgo.By("Deleting the first pod")
+				err = e2epod.DeletePodWithWait(m.cs, pod)
+				framework.ExpectNoError(err, "while deleting the first pod")
```

Comment:
```suggestion
				framework.ExpectNoError(err, "failed to delete the first pod")
```

Reply:
Fixed

Reply:
These additional strings are tricky. I am currently updating `[writing-good-e2e-tests.md](https://github.com/kubernetes/community/pull/7021). So far, I am not covering this aspect, but perhaps I should. Let's discuss here so that you can tell me what you think with a specific example.

First, these are failure messages, which means that they will be presented in a context where it is clear that something failed, for example with the big `[FAILED]` in front of this text. Therefore `failed to` is redundant.

Second, it's a bit like error wrapping in Go: if every place where an error gets wrapped adds "failed to", we end up with `failed to do a: failed to do b: failed to do c` - not very useful.

Therefore I prefer to avoid it entirely.

The `while` also looks like a fill word that could be dropped. `deleting the first pod` seems sufficient to me to describe the operation.

Reply:
I rephrased the errors, PTAL.